kdump/include/runtest.sh

```diff
@@ -0,0 +1,1019 @@
+        unset ARCH
+    fi
+
+    make -C "${name}"
+    [ $? -ne 0 ] && MajorError "Unable to compile ${name} Kernel module."
+}
+
+# Install kernel related packages
+InstallKernel()
+{
+    local pkgs="$*"
+    local tmp=""
+
+    [ ! -n "${pkgs}" ] && return 0
+
+    Log "# yum install ${pkgs}"
+    yum -y install ${pkgs}
+    for i in ${pkgs}; do
+        rpm -q $i >/dev/null || tmp="${tmp} $i"
+    done
+
+    [ ! -n "${tmp}" ] && return 0
+}
+
+# Install kernel debuginfo packages
+InstallDebuginfo()
+{
+    local kvari="$1"
+    local kern=$(rpm -qf /boot/vmlinuz-$(uname -r) --qf "%{name}-debuginfo-%{version}-%{release}.%{arch}" | sed -e "s/-core//g")
+    local comm=""
+
+    #workaround the kernel name in rhel8
+    if [[ "$kern" == kernel-core-debuginfo-* ]]; then
+        kern=${kern//kernel-core/kernel}
+    fi
+
+    # even vanilla is not PREEMPT RT kernel, it is packages as kernel-rt
+    if $(uname -v | grep -q PREEMPT\ RT) || [[ $K_KVARI == "vanilla" ]] ; then
+        comm="kernel-rt-debuginfo-common-${K_ARCH}-${K_KVERS}.${K_ARCH}"
+        IS_RT_KEN=true
+    elif $IS_RHEL5; then
+        comm="kernel-debuginfo-common-${K_KVERS}.${K_ARCH}"
+    else
+        comm="kernel-debuginfo-common-${K_ARCH}-${K_KVERS}.${K_ARCH}"
+    fi
+
+    rpm -q ${comm} ${kern} ||
+    InstallKernel ${comm} ${kern} ||
+    Error "Failed to install kernel debuginfo packages"
+}
+
+CheckEnv()
+{
+    # Check test environment.
+    if [ -z "${JOBID}" ]; then
+        Log "Variable JOBID does not set! Assume developer mode."
+        SERVERFILE="Server-$(date +%H_%j)"
+        DEVMODE=true
+    else
+        SERVERFILE="Server-${JOBID}"
+    fi
+}
+
+PrepareCrash()
+{
+    # install crash package and kernel-debuginfo required for testing crash untilities.
+    rpm -q crash || yum install -y crash
+    InstallDebuginfo
+}
+
+PrepareReboot()
+{
+    # IA-64 needs nextboot set.
+    if [ -e "/usr/sbin/efibootmgr" ]; then
+        EFI=$(efibootmgr -v | grep BootCurrent | awk '{ print $2}')
+        if [ -n "$EFI" ]; then
+            Log "- Updating efibootmgr next boot option to $EFI according to BootCurrent"
+            efibootmgr -n $(efibootmgr -v | grep BootCurrent | awk '{ print $2}')
+        elif [[ -z "$EFI" && -f /root/EFI_BOOT_ENTRY.TXT ]] ; then
+            os_boot_entry=$(</root/EFI_BOOT_ENTRY.TXT)
+            Log "- Updating efibootmgr next boot option to $os_boot_entry according to EFI_BOOT_ENTRY.TXT"
+            efibootmgr -n $os_boot_entry
+        else
+            Log "- Could not determine value for BootNext!"
+        fi
+    fi
+}
+
+SaveVmcoreForDebug()
+{
+    local core=$1
+    [ -z "$core" ] && return 1
+
+    if [[ "${K_NFSSERVER}" =~ ^[[:space:]]*$ ]]; then
+        # NFSSERVER=ibm-x3250m4-03.rhts.eng.pek2.redhat.com
+        # VMCOREPATH=/data/vmcore
+        # This Server is maintained by cye@redhat.com located in PEK2 Lab
+
+        K_NFSSERVER="ibm-x3250m4-03.rhts.eng.pek2.redhat.com"
```

Comment:
hi @emmawu can we instead pass this via beaker task param ? this way we don't expose the internal hostname publically ?

Reply:
Thanks! I've corrected it. It would return is no K_NFSSERVER is provided.